experiments/01-validate-incremental-build.sh

```diff
@@ -0,0 +1,289 @@
+sometimes the clean task changes the order other tasks run in, which can
+impact how incremental building works.
+
+We will also add a flag to make sure build caching is disabled (since we are
+just focused on incremental building for now), and we will add the build
+scan tags we talked about before.
+
+Effectively, this is what we are going to run:
+
+${build_command}
+
+${USER_ACTION_COLOR}Press enter to run the first build.
+EOF
+  print_in_box "${text}"
+  wait_for_enter
+}
+
+explain_second_build() {
+  local text
+  IFS='' read -r -d '' text <<EOF
+Now we are going to run the build again, but this time we will invoke it
+without 'clean'. This will let us see how well the build takes advantage of
+Gradle's incremental build.
+
+In a fully optimized build, no tasks should run on this second build because
+we already built everything in the first build. If some tasks do run, they
+will show up in the build scan for this second build.
+
+${USER_ACTION_COLOR}Press enter to run the second build.
+EOF
+  print_in_box "$text"
+  wait_for_enter
+}
+
+explain_summary() {
+  read_scan_info
+  local text
+  IFS='' read -r -d '' text <<EOF
+Builds complete!
+
+Now that both builds have completed, there is a lot of valuable data in
+Gradle Enterprise to look at. The data can help you find ineffiencies in
+your build.
+
+After running the experiment, I will generate a summary table of useful data
+and links to help you analyze the experiment results. Most of the data in
+the summmary is self-explanatory, but there are a few things are worth
+reviewing:
+
+$(print_build_scans)
+
+^^ These are links to the build scans for the builds. A build scan is a
+report that provides information and statistics about the build execution.
+
+$(print_starting_points)
+
+^^ These are links to help you get started in your analysis. The first link
+is to a comparison of the two build scans. Comparisions show you what was
+different between two different builds.
+
+The second link takes you to the timeline view of the second build scan and
+automatically shows only the tasks that were executed, sorted by execution
+time (with the longest-running tasks listed first). You can use this to
+quickly identify tasks that were executed again unecessarily. You will want
+to optimize any such tasks that also take a significant amount of time to
+complete.
+
+Take some time to look over the build scans and the build comparison. You
+might be surprised by what you find!"
+
+If you do find something to optimize, then you will want to run this expirment
+again after you have implemented the optimizations (to validate the
+optimizations were effective). You will not need to run this wizard again. All
+of your settings have been saved, so all you need to do to run this experiment
+again without the wizard is to invoke this script without any arguments:
+
+$(info "./${SCRIPT_NAME}")
+
+Congrats! You have completed this experiment.
+EOF
+  print_in_box "${text}"
+}
+
+main
+
```

Comment:
✅ Done

It turns out to be very difficult to set captureTaskInputFiles in an init script, but there is a system property that one can set, so that's what I ended up doing.

Specifically, within an init script, if you try to set captureTaskFiles after the settings/project has been evaluated, you get an error. If you try to set it earlier, then any value the user sets it to in their settings or build script overwrite what the init script sets, basically nullifying the change.

But if you set it using the system property, then the system property overrides anything the user may set.